rkt/list.go

```diff
@@ -102,10 +134,115 @@ func runList(cmd *cobra.Command, args []string) (exit int) {
+	lines := []string{
+		fmt.Sprintf("Unable to load pod %s manifest, because it is invalid:", p.uuid.String()),
+		fmt.Sprintf("  %v", err),
+	}
+	pm := lastditch.PodManifest{}
+	if err := pm.UnmarshalJSON(pmj); err != nil {
+		lines = append(lines, "  Also, failed to get any information about invalid pod manifest:")
+		lines = append(lines, fmt.Sprintf("    %v", err))
+	} else {
+		if len(pm.Apps) > 0 {
+			lines = append(lines, "Objects related to this error:")
+			for _, app := range pm.Apps {
+				lines = append(lines, fmt.Sprintf("  %s", appLine(app)))
+			}
+		} else {
+			lines = append(lines, "No other objects related to this error")
+		}
+	}
+	return fmt.Errorf("%s", strings.Join(lines, "\n"))
+}
+
+func newPodListZeroAppsError(p *pod) error {
+	return fmt.Errorf("Pod %s contains zero apps", p.uuid.String())
+}
+
+func newPodListImageStoreFailure(p *pod, err error, pm *schema.PodManifest, app schema.RuntimeApp) error {
+	lines := []string{
+		fmt.Sprintf("Unable to get image %s manifest from store:", app.Image.ID.String()),
+		fmt.Sprintf("  %v", err),
+		"Objects related to this error:",
+		fmt.Sprintf("  App: %q from failing image %q (%s)",
+			app.Name, app.Image.Name, app.Image.ID),
+		fmt.Sprintf("  Pod %s (spec version %s) with following apps:", p.uuid.String(), pm.ACVersion.String()),
+	}
+	for _, pApp := range pm.Apps {
+		lines = append(lines, fmt.Sprintf("    %s", appLine(degradeRuntimeApp(pApp))))
+	}
+	return fmt.Errorf("%s", strings.Join(lines, "\n"))
+}
+
+func newPodListImageLoadFailure(p *pod, err error, pm *schema.PodManifest, imj []byte, app schema.RuntimeApp) error {
+	im := lastditch.ImageManifest{}
+	imErr := im.UnmarshalJSON(imj)
+	acVersion := "unknown"
+	if imErr == nil {
+		acVersion = im.ACVersion
+	}
+	lines := []string{
+		fmt.Sprintf("Unable to load image %s manifest (spec version %s) because it is invalid:", app.Image.ID.String(), acVersion),
+		fmt.Sprintf("  %v", err),
+	}
+	if imErr != nil {
+		lines = append(lines, "  Also, failed to get any information about invalid image manifest:")
+		lines = append(lines, fmt.Sprintf("    %v", imErr))
+	}
+	lines = append(lines, "Objects related to this error:")
+	lines = append(lines, fmt.Sprintf("  App: %q from failing image %q (%s)",
+		app.Name, app.Image.Name, app.Image.ID))
+	lines = append(lines, fmt.Sprintf("  Pod %s (spec version %s) with following apps:", p.uuid.String(), pm.ACVersion.String()))
+	for _, pApp := range pm.Apps {
+		lines = append(lines, fmt.Sprintf("    %s", appLine(degradeRuntimeApp(pApp))))
+	}
+	return fmt.Errorf("%s", strings.Join(lines, "\n"))
+}
+
+func degradeRuntimeApp(app schema.RuntimeApp) lastditch.RuntimeApp {
+	return lastditch.RuntimeApp{
+		Name: app.Name.String(),
+		Image: lastditch.Image{
+			Name: app.Image.Name.String(),
+			ID:   app.Image.ID.String(),
+		},
+	}
+}
+
+func appLine(app lastditch.RuntimeApp) string {
+	return fmt.Sprintf("App: %q from image %q (%s)",
+		app.Name, app.Image.Name, app.Image.ID)
```

Comment:
`app.Image.ID` should be printed by following the flag `--full`: see [image_list.go#L215](https://github.com/coreos/rkt/blob/master/rkt/image_list.go#L215) and #1455